src/System.IO/ref/System.IO.cs

```diff
@@ -162,13 +170,17 @@ public abstract partial class Stream : System.IDisposable
         public abstract long Position { get; set; }
         public virtual int ReadTimeout { get { return default(int); } set { } }
         public virtual int WriteTimeout { get { return default(int); } set { } }
+        public virtual System.IAsyncResult BeginRead(byte[] buffer, int offset, int count, System.AsyncCallback callback, object state) { return default(System.IAsyncResult); }
```

Comment:
Agreed, we decided to not hide it. We did decide in principle that we wanted to change the base implementation to delegate, but as you point out that's not a complete answer.
Hiding it later would not be a breaking change, and future tooling/analyzers can help steer customers away from the old APM.